lib/omniauth/strategies/openid_connect.rb

```diff
@@ -312,13 +317,36 @@ def id_token_callback_phase
         call_app!
       end
 
+      def type_response?(type)
+        if options.response_type.is_a?(Array)
```

Comment:
This code looks very familiar to me. Does it duplicate `only_id_token?` for at least 50%?

Please consider extracting `response_type` normalization to separate instance method and get rid of duplication:

```ruby
def response_type
  @response_type ||= Array(options.response_type).map(&:to_s)
end

# not sure if we really need this method now
def type_response?(type)
  response_type.include?(type.to_s)
end

def only_id_token?
  response_type == ['id_token']
  # or
  # response_type.include?('id_token') && response_type.size == 1
end
```